### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by makermelissa**

### CODE REVIEW

**Comment by makermelissa** [comment]
Tagging @ladyada

File: src/NMEA_build.cpp [hunk 1]
```diff
@@ -565,7 +565,13 @@ char *Adafruit_GPS::build(char *nmea, const char *thisSource,
 
   addChecksum(nmea); // Successful completion
   if (!noCRLF) { // Add Carriage Return and Line Feed to comply with NMEA-183
-    sprintf(nmea, "%s\r\n", nmea);
+    size_t len = strlen(nmea);
+    char *newStr =
+        (char *)malloc(len + 3); // +2 for \r\n, +1 for null terminator
+    if (newStr) {
+      strcpy(newStr, nmea);   // Copy original string
+      strcat(newStr, "\r\n"); // Append \r\n
```
Comment (ladyada) [inline src/NMEA_build.cpp]: newstr is never used after this and we return nmea so this code does not actually append \r\n to nmea?

Comment (makermelissa) [reply src/NMEA_build.cpp]: Thanks, I'll take another look at the code.

**Comment by makermelissa** [comment]
@ladyada could I get another review? Thanks.

File: src/NMEA_build.cpp [hunk 2]
```diff
@@ -565,7 +565,14 @@ char *Adafruit_GPS::build(char *nmea, const char *thisSource,
...
   addChecksum(nmea); // Successful completion
   if (!noCRLF) { // Add Carriage Return and Line Feed to comply with NMEA-183
-    sprintf(nmea, "%s\r\n", nmea);
+    size_t len = strlen(nmea);
+    char *nmeaWithCRLF =
+        (char *)malloc(len + 3); // +2 for \r\n, +1 for null terminator
+    if (nmeaWithCRLF) {
+      strcpy(nmeaWithCRLF, nmea);   // Copy original string
+      strcat(nmeaWithCRLF, "\r\n"); // Append \r\n
+      return nmeaWithCRLF;          // return pointer to finished product
```
Comment (ladyada) [inline src/NMEA_build.cpp]: we need to return the same buffer, so strcpy it back and free the temp buff

**Review by ladyada — CHANGES_REQUESTED** [review]

**Comment by makermelissa** [comment]
I added the suggested fix, but didn't tag you again @ladyada.

**Review by ladyada — APPROVED** [review]

**Pull request merged**